Submit-PolicyDefinitionFile.ps1

```diff
@@ -0,0 +1,306 @@
+#Requires -PSEdition Core
+
+param(
+    [parameter(Mandatory = $true, Position = 0)] $fileName,
+    [parameter(Mandatory = $false, Position = 1)] [string] $outputDirectory = $null,
+    [switch] $skipFileSplitting
+)
+
+$ErrorActionPreference = "Stop"
+
+$files = Get-ChildItem -Path $fileName
+if ($files.Count -eq 0) {
+    Write-Error "File not found: $fileName."
+}
+elseif ($files.Count -gt 1) {
+    Write-Error "Multiple files found: $fileName. Specify a file, not a directory"
+}
+$file = $files[0]
+$content = Get-Content $file.FullName -Raw
+if (!(Test-Json $content)) {
+    Write-Error "File is not valid JSON: $fileName."
+}
+$definition = ConvertFrom-Json $content -AsHashtable -Depth 100
+
+# analyze structure
+$properties = $definition
+if ($definition.properties) {
+    $properties = $definition.properties
+}
+
+$name = $definition.name
+if (!$name) {
+    if ($definition.id) {
+        $idSplits = $definition.id -split "/"
+        $name = $idSplits[-1]
+    }
+    elseif ($properties.id) {
+        $idSplits = $properties.id -split "/"
+        $name = $idSplits[-1]
+    }
+}
+if ($name) {
+    if (!([guid]::TryParse($name, $([ref][guid]::Empty)))) {
+        $oldName = $name
+        $name = (New-Guid).Guid
+        Write-Warning "name $oldName not a GUID. Using generated GUID $name as the name"
+    }
+}
+else {
+    $name = (New-Guid).Guid
+    Write-Warning "name missing. Using generated GUID $name as the name"
+}
+
+$displayName = $properties.displayName
+if (!$displayName) {
+    Write-Error "displayName not found."
+}
+$description = $properties.description
+if (!$description) {
+    Write-Error "description not found."
+}
+
+$version = $properties.version
+# if (!$version) {
```

Comment:
Why leave commented-out code in?

Not following logic here:
$version = $properties.version (doesn't exist)

If version is missing, it's not added to the metadata property.

Reply:
This is needed due to the current state of version support by PG. This is about the new version, not the version in metadata.
Would you want a check on the metadata for now?

Reply:
Yes, let's